doc/hotspot-style.md

```diff
@@ -572,9 +572,13 @@ There are a few exceptions to this rule.
 
 * `#include <new>` to use placement `new`, `std::nothrow`, and `std::nothrow_t`.
 * `#include <limits>` to use `std::numeric_limits`.
-* `#include <type_traits>`.
+* `#include <type_traits>` with some restrictions, listed below.
 * `#include <cstddef>` to use `std::nullptr_t` and `std::max_align_t`.
 
+Certain restrictions apply to the declarations provided by `<type_traits>`.
+
+* The `alignof` operator should be utilized rather than `std::alignment_of<>`.
```

Comment:
s/utilized/used/
https://scientistseessquirrel.wordpress.com/2019/04/16/for-the-love-of-all-that-is-holy-stop-writing-utilize/
:)

Reply:
And here I was thinking this would sound more professional :P

Speaking of which, could I include a condensed and shortened version of this part from our discussion above? I've left it out in the meantime but just wanted to ask in case
> std::alignment_of<> could be used in advanced metaprogramming contexts
involving higher order metafunctions. We don't do that sort of thing much at
all in HotSpot. If someone finds a place where std::alignment_of<> might be
better, we can revisit its status.

Reply:
I don't think that's needed. We already have a blanket statement regarding
violations: "Failure to follow these guidelines may lead to discussion during
code reviews, if not outright rejection of a change." Anyone attempting to add
such code is presumably capable of defending it.